myuw/management/commands/clear_expired_sessions.py

```diff
@@ -0,0 +1,31 @@
+"""
+The django clearsessions commend internally calls:
+   cls.get_model_class().objects.filter(
+       expire_date__lt=timezone.now()).delete()
+which could lock the DB table for a long time when
+ having a large number of records to delete.
+
+To prevent the job running forever, we only delete 40000
+expired django sessions in a single run
+"""
+
+import logging
+from django.core.management.base import BaseCommand, CommandError
+from django.contrib.sessions.models import Session
+from django.utils import timezone
+
+logger = logging.getLogger(__name__)
+DEL_SIZE = 40000
+
+
+class Command(BaseCommand):
+
+    def handle(self, *args, **options):
+        i = 0
+        for record in Session.objects.filter(expire_date__lt=timezone.now()):
```

Comment:
What about using the built in QuerySet delete; that's supposed to do them in bulk?  
https://docs.djangoproject.com/en/3.0/topics/db/queries/#deleting-objects
The DEL_SIZE limit could still be achieved by limiting the results in the Session query

Reply:
Thank you! Updated.